wgpu-core/src/command/bundle.rs

```diff
@@ -1036,14 +1040,20 @@ pub mod bundle_ffi {
         stages: wgt::ShaderStage,
         offset: u32,
         size_bytes: u32,
-        data: *const u32,
+        data: *const u8,
     ) {
         span!(_guard, DEBUG, "RenderBundle::set_push_constants");
-        let data_slice = slice::from_raw_parts(data, (size_bytes / 4) as usize);
+        let data_slice = slice::from_raw_parts(data, size_bytes as usize);
         let value_offset = pass.base.push_constant_data.len().try_into().expect(
             "Ran out of push constant space. Don't set 4gb of push constants per RenderBundle.",
         );
+
+        let aligned_len = ((data_slice.len() + 3) / 4) * 4;
+        pass.base.push_constant_data.reserve(aligned_len);
```

Comment:
Instead of changing internal data to use u8s, we should convert to u32s in this step. As we are copying anyway, we can use a loop of (something like):

```rust
base.push_constant_data.extend(data_slice.chunks_exact(4).map(|arr| {
    u32::from_ne_bytes([
        arr[0],
        arr[1],
        arr[2],
        arr[3],
    ])
}));
```

Then assert somewhere (probably wgpu-rs) that size is a multiple of four. There are ways of doing this that allows it to be not a multiple of 4, but all types in glsl are of size 4 anyway, so I don't think that's something we need to support.

Reply:
I'm pondering whether I should pad unaligned data, or I just assert it to panic; and it should be noted that [16 bit types](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_KHR_16bit_storage.html) or even [8 bit types](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_KHR_8bit_storage.html) are allowed in GLSL and have been used quite extensively on mobile platforms. So I think it is something worth dealing with in this crate.